scripts/generate_mrenclave.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ "$1" == "" ]; then
+    echo 'Parse the output of the SGX tool to extract the MRENCLAVE value'
+    echo 'and save it to file $TCF_HOME/metadata_info.txt by default'
+    echo 'Usage: generate_mrenclave.sh <library location> <output file(optional)>'
+    exit
+fi
+
+LIB="$1"
+if [ -z "$2" ];then
+	OUT_FILE="${TCF_HOME}/metadata_info.txt"
+else
+	OUT_FILE=$2
+fi	
+
+sgx_sign dump -enclave ${LIB} -dumpfile metadata.txt
+
+awk '/metadata->enclave_css.body.enclave_hash.m:/{flag=1; next} 
+    /metadata->enclave_css.body.isv_prod_id:/{flag=0} flag' metadata.txt | awk '!a[$0]++' > ${OUT_FILE}
+
```

Comment:
Need to extract 32 bytes. MRECNLAVE value is 32 bytes (64 hex characters). So need both lines after `metadata->enclave_css.body.enclave_hash.m` and then combine both as hex string (first line hex string + " " + second line hex string)
```
metadata->enclave_css.body.enclave_hash.m:
0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00
0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00
```

Reply:
@karthikamurthy Please try below. This will save MRENCLAVE value as 32 bytes (as hex values) in out file with space as delimiter for each bytes.

```
sgx_metadata="sgx_sign dump -enclave ${LIB} -dumpfile /dev/stdout"
mrenclave_search_pattern="/metadata->enclave_css.body.enclave_hash.m:/{flag=1; next} \
        /metadata->enclave_css.body.isv_prod_id:/{flag=0} flag"
first_two_lines="head -n2"
merge_lines="sed N;s/\n//"
strip_trailing_space="sed -e s/[[:blank:]]*$//"

mrenclave_hex_bytes="$(awk "$mrenclave_search_pattern" <($sgx_metadata) |\
        ($first_two_lines) | ($merge_lines) | ($strip_trailing_space))"

echo $mrenclave_hex_bytes
echo $mrenclave_hex_bytes > ${OUT_FILE}
```
The python code below can use the metadata_info.txt to convert MRENCALVE value to hex string as below.

```
file = open("metadata_info.txt","r")
hexlist = file.read().rstrip('\r\n ').split(' ')
hexbytes = bytes([int(x,0) for x in hexlist])
hexstring = hexbytes.hex()
print(hexstring)
```